### ISSUE TITLE
Bug: Cannot load any private message

### ISSUE THREAD

**Issue by jandamm** [Dimillian/IceCubesApp#696]
I can't show any private messages, I always get a fullscreen error message.
Tapping retry doesn't help.

![image](https://user-images.githubusercontent.com/5963139/217186650-36f80679-49a2-4356-a33d-adb13a949026.png)

I have some non linear private messages. So not every message was a reply to the oldest message. Some private mentions have multiple replies - like normal messages. Maybe this is the issue.
Showing those messages in Mammoth results in incomplete threads (some are filtered out)

**Comment by frepke** [comment]
Have almost the same problem, but the first time after opening the app, the messages are displayed (still empty for me). The error occurs after a refresh and never goes away until I restart the app again.

**Comment by jandamm** [comment]
Now I got the first private mention after using Ice Cubes.
Now this conversation shows in the UI. But the others won't.
In case it's not possible to find all conversations there should at least be a message instead of this error.

**Comment by sothawo** [comment]
I can only see the last message in this tab. When a new one comes in, I see this new one, but not the older ones
Running version 1.4.1

**Comment by sothawo** [comment]
I found the reason, a nullability issue when loading the conversations.

**Comment by jandamm** [comment]
I just updated to 1.4.2 and it's broken again 😅

**Comment by sothawo** [comment]
I meant I am running version 1.4.2 and have this error, but I'm just on fixing it.

**Comment by frepke** [comment]
Error still exists in version 1.4.4 :(

**Comment by jandamm** [comment]
Seems to be another reason.
For me it was fixed with 1.4.3.

**Comment by frepke** [comment]
It's only when de inbox is empty. One message is enough to solve this problem.

**Comment by Dimillian** [comment]
This is what you're supposed to see if it's empty 
![image](https://user-images.githubusercontent.com/535509/218090100-59ccbd8e-2304-497f-9acc-cf6f6a0996bb.png)

**Comment by frepke** [comment]
Yes, I know. This is wat I see when restarting the app. But when I refresh this page (pull down) it results in the error again.

**Comment by sothawo** [comment]
I cannot reproduce it in the simulator,there the refresh works. When debugging on the device, I see the following error on refresh:

```
Update NavigationAuthority bound path tried to update multiple times per frame.
Error Domain=NSURLErrorDomain Code=-999 "cancelled" UserInfo={NSErrorFailingURLStringKey=https://infosec.exchange/api/v1/conversations, NSErrorFailingURLKey=https://infosec.exchange/api/v1/conversations, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <0429CED2-5B2F-454E-8DA2-EC0E4602338E>.<27>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <0429CED2-5B2F-454E-8DA2-EC0E4602338E>.<27>, NSLocalizedDescription=cancelled}
```
Information I found either show a problem with certificates - which I don't think is the problem here, we have calls before and after that that succeed - or something "executing the exact same request while the same object is already executing - which I could not validate

Currently out of ideas

**Comment by Dimillian** [comment]
Feel like it's iOS .refreshable canceling the request. Are you on 16.3? If not can you update.

**Comment by sothawo** [comment]
yes, 16.3

**Comment by sothawo** [comment]
strange thing is, that refreshes on all other tabs are working, only the ConversationsListView has this behaviour

**Comment by frepke** [comment]
> yes, 16.3

I'm on 16.3 also.

**Comment by sothawo** [comment]
found it.

The call in the .refreshable need to be wrapped in a `Task`:

@Dimillian I'll add this to the NotificationsListView and TimeLineView as well, there the await methods for fetching data in the model are missing this as well, PR incoming

**Comment by Dimillian** [comment]
That the thing, it should not requiere that. .refreshable is already a Task. And that how it sync the status of the refresh. It you wrap it then you loose the completion.

**Comment by Dimillian** [comment]
https://www.swiftbysundell.com/articles/making-swiftui-views-refreshable/

**Comment by sothawo** [comment]
ok, I see in the docs of refreshable that there is no Task wrapping as well. But why does it fail then?

**Comment by Dimillian** [comment]
I think it's just a SwiftUI bug unfortunately and this screen must trigger it. I'm ok to wrap it for this one tho if it fix it. Never had it on my phone.

**Comment by frepke** [comment]
Problem is solved in 1.4.6 but still has a weird glitch: https://user-images.githubusercontent.com/6153559/218331677-aa898c4d-b01e-431d-8018-e575c319c892.MOV
